### PR TITLE
Update bloom filter

### PR DESCRIPTION
**Pull request by rllola**
Everytime we generate a new address we should update the filter.

### CODE REVIEW

**Comment by xanimo** [comment]
ACK. tested on x86_64-pc-linux-gnu. took a bit to setup node and if not otherwise noted i think we should emphasize users using nodejs LTS v16. bc i had tried with non LTS version and encountered an error. also are you aware of this?
```
Error [NotFoundError]: Key not found in database [0ed1884ab0193d74f5f8a7f674808f7365ceeba6]
    at /mnt/source/repos/dogecoin-spv-node/node_modules/levelup/lib/levelup.js:188:15
    at /mnt/source/repos/dogecoin-spv-node/node_modules/encoding-down/index.js:75:21 {
  [cause]: [Error: NotFound: ]
}
```

i will try to update packages and see if it resolves anything but other than that it's performing as you explained to me.

File: test/integration/issue4.test.js [hunk 1]
```diff
@@ -79,4 +78,4 @@ test.serial('should register paymentchannel tx after deleting database', async t
     await new Promise(resolve => setTimeout(resolve, 1000))
   
     t.pass()
-  })
\ No newline at end of file
+})
```
Comment (xanimo) [inline test/integration/issue4.test.js:81]: new line

File: test/integration/resync.test.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  await spvnode.synchronize()
+
+  await new Promise(resolve => setTimeout(resolve, 1000))
+
+  await spvnode.shutdown()
+
+  const balance = await wallet.getBalance()
+
+  t.is(balance, 600000000000n)
+})
```
Comment (xanimo) [inline test/integration/resync.test.js:76]: new line

File: test/integration/updatefilter.test.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  await spvnode.synchronize()
+
+  await new Promise(resolve => setTimeout(resolve, 1000))
+
+  await spvnode.shutdown()
+
+  const balance = await wallet.getBalance()
+
+  t.is(balance, 315000000000n)
+})
```
Comment (xanimo) [inline test/integration/updatefilter.test.js:135]: new line

**Review by xanimo — APPROVED** [review]
ACK despite error below (expected?) and lack of new lines. tested on x86_64-pc-linux-gnu. are you aware of this?
```
Error [NotFoundError]: Key not found in database [0ed1884ab0193d74f5f8a7f674808f7365ceeba6]
    at /mnt/source/repos/dogecoin-spv-node/node_modules/levelup/lib/levelup.js:188:15
    at /mnt/source/repos/dogecoin-spv-node/node_modules/encoding-down/index.js:75:21 {
  [cause]: [Error: NotFound: ]
}
```
took a bit to setup nodejs and remove broken yarn apt repository/gpg key stuff. if not otherwise noted i think we should emphasize users using nodejs LTS v16. bc i had tried with non LTS version v16 and encountered an error. 

i will try to update packages and see if it resolves anything but other than that it's performing as you explained to me.

**Comment by rllola** [comment]
The error that you add should have been fixed when this was closed.
https://github.com/ShibeTechnology/dogecoin-spv-node/issues/4

I am going to try to sync again and play with it. Let's see if it happens again.

**Pull request merged**